The job **`aarch64-gnu-llvm-21-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/28127528863/job/83295200941) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/83295200941) [(plain)](https://github.com/rust-lang/rust/commit/494adf0bf0a784cec8e90a051a85052f15d865c5/checks/83295200941/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Saved the actual stderr to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/asm/parse-error/parse-error.stderr`
diff of stderr:

57    |
58 LL |         asm!("{}", in(reg) foo => bar);
59    |                                ^^ expected one of 7 possible tokens
+    |
+ help: you might have meant to write a "greater than or equal to" comparison
+    |
+ LL -         asm!("{}", in(reg) foo => bar);
+ LL +         asm!("{}", in(reg) foo >= bar);
+    |
60 
61 error: expected a path for argument to `sym`
62   --> $DIR/parse-error.rs:32:24


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args asm/parse-error.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/asm/parse-error.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/asm/parse-error" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: requires at least a template string argument
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:12:9
   |
---

error: expected `)`, found `foo`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:22:27
   |
LL |         asm!("{}", in(reg foo));
   |                           ^^^ expected `)`

error: expected expression, found end of macro arguments
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:24:27
   |
LL |         asm!("{}", in(reg));
   |                           ^ expected expression

error: expected register class or explicit register
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:26:26
   |
LL |         asm!("{}", inout(=) foo => bar);
   |                          ^

error: expected expression, found end of macro arguments
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:28:37
   |
LL |         asm!("{}", inout(reg) foo =>);
   |                                     ^ expected expression

error: expected one of `!`, `,`, `.`, `::`, `?`, `{`, or an operator, found `=>`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:30:32
   |
LL |         asm!("{}", in(reg) foo => bar);
   |                                ^^ expected one of 7 possible tokens
   |
help: you might have meant to write a "greater than or equal to" comparison
   |
LL -         asm!("{}", in(reg) foo => bar);
LL +         asm!("{}", in(reg) foo >= bar);
   |

error: expected a path for argument to `sym`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:32:24
   |
LL |         asm!("{}", sym foo + bar);
   |                        ^^^^^^^^^

error: expected one of `)`, `att_syntax`, `may_unwind`, `nomem`, `noreturn`, `nostack`, `preserves_flags`, `pure`, `raw`, or `readonly`, found `foo`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:34:26
   |
LL |         asm!("", options(foo));
   |                          ^^^ expected one of 10 possible tokens

error: expected one of `)` or `,`, found `foo`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:36:32
   |
LL |         asm!("", options(nomem foo));
   |                                ^^^ expected one of `)` or `,`

error: expected one of `)`, `att_syntax`, `may_unwind`, `nomem`, `noreturn`, `nostack`, `preserves_flags`, `pure`, `raw`, or `readonly`, found `foo`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:38:33
   |
LL |         asm!("", options(nomem, foo));
   |                                 ^^^ expected one of 10 possible tokens

error: at least one abi must be provided as an argument to `clobber_abi`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:45:30
   |
LL |         asm!("", clobber_abi());
   |                              ^

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:47:30
   |
LL |         asm!("", clobber_abi(foo));
   |                              ^^^ not a string literal

error: expected one of `)` or `,`, found `foo`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:49:34
   |
LL |         asm!("", clobber_abi("C" foo));
   |                                  ^^^ expected one of `)` or `,`

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:51:35
   |
LL |         asm!("", clobber_abi("C", foo));
   |                                   ^^^ not a string literal

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:53:30
   |
LL |         asm!("", clobber_abi(1));
   |                              ^ not a string literal

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:55:30
   |
LL |         asm!("", clobber_abi(()));
   |                              ^ not a string literal

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:57:30
   |
LL |         asm!("", clobber_abi(uwu));
   |                              ^^^ not a string literal

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:59:30
   |
LL |         asm!("", clobber_abi({}));
   |                              ^ not a string literal

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:61:30
   |
LL |         asm!("", clobber_abi(loop {}));
   |                              ^^^^ not a string literal

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:63:30
   |
LL |         asm!("", clobber_abi(if));
   |                              ^^ not a string literal

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:65:30
   |
LL |         asm!("", clobber_abi(do));
   |                              ^^ not a string literal

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:67:30
   |
LL |         asm!("", clobber_abi(<));
   |                              ^ not a string literal

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:69:30
   |
LL |         asm!("", clobber_abi(.));
   |                              ^ not a string literal

error: duplicate argument named `a`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:77:36
   |
LL |         asm!("{a}", a = const foo, a = const bar);
   |                     -------------  ^^^^^^^^^^^^^ duplicate argument
   |                     |
   |                     previously here

error: argument never used
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:77:36
   |
LL |         asm!("{a}", a = const foo, a = const bar);
   |                                    ^^^^^^^^^^^^^ argument never used
   |
   = help: if this argument is intentionally unused, consider using it in an asm comment: `"/* {1} */"`

error: expected one of `#`, `clobber_abi`, `const`, `in`, `inlateout`, `inout`, `label`, `lateout`, `options`, `out`, or `sym`, found `""`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:83:29
   |
LL |         asm!("", options(), "");
   |                             ^^ expected one of 11 possible tokens

error: expected one of `#`, `clobber_abi`, `const`, `in`, `inlateout`, `inout`, `label`, `lateout`, `options`, `out`, or `sym`, found `"{}"`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:85:33
   |
LL |         asm!("{}", in(reg) foo, "{}", out(reg) foo);
   |                                 ^^^^ expected one of 11 possible tokens

error: asm template must be a string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:87:14
   |
LL |         asm!(format!("{{{}}}", 0), in(reg) foo);
   |              ^^^^^^^^^^^^^^^^^^^^

error: asm template must be a string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:89:21
   |
LL |         asm!("{1}", format!("{{{}}}", 0), in(reg) foo, out(reg) bar);
   |                     ^^^^^^^^^^^^^^^^^^^^

error: _ cannot be used for input operands
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:91:28
   |
LL |         asm!("{}", in(reg) _);
   |                            ^

error: _ cannot be used for input operands
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:93:31
   |
LL |         asm!("{}", inout(reg) _);
   |                               ^

error: _ cannot be used for input operands
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:95:35
   |
LL |         asm!("{}", inlateout(reg) _);
   |                                   ^

error: requires at least a template string argument
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:102:1
   |
LL | global_asm!();
   | ^^^^^^^^^^^^^

error: asm template must be a string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:104:13
   |
LL | global_asm!(FOO);
   |             ^^^

error: expected token: `,`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:106:18
   |
LL | global_asm!("{}" FOO);
   |                  ^^^ expected `,`

error: expected operand, options, or additional template string
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:108:19
   |
LL | global_asm!("{}", FOO);
---
   |
LL | global_asm!("{}", const);
   |                        ^ expected expression

error: expected one of `,`, `.`, `?`, or an operator, found `FOO`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:112:30
   |
LL | global_asm!("{}", const(reg) FOO);
   |                              ^^^ expected one of `,`, `.`, `?`, or an operator

error: expected one of `)`, `att_syntax`, or `raw`, found `FOO`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:114:25
   |
LL | global_asm!("", options(FOO));
   |                         ^^^ expected one of `)`, `att_syntax`, or `raw`

error: expected one of `)`, `att_syntax`, or `raw`, found `FOO`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:116:25
   |
LL | global_asm!("", options(FOO,));
   |                         ^^^ expected one of `)`, `att_syntax`, or `raw`

error: expected one of `)` or `,`, found `FOO`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:118:31
   |
LL | global_asm!("", options(nomem FOO));
   |                               ^^^ expected one of `)` or `,`

error: expected one of `)`, `att_syntax`, or `raw`, found `FOO`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:120:32
   |
LL | global_asm!("", options(nomem, FOO));
   |                                ^^^ expected one of `)`, `att_syntax`, or `raw`

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:123:29
   |
LL | global_asm!("", clobber_abi(FOO));
   |                             ^^^ not a string literal

error: expected one of `)` or `,`, found `FOO`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:125:33
   |
LL | global_asm!("", clobber_abi("C" FOO));
   |                                 ^^^ expected one of `)` or `,`

error: expected string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:127:34
   |
LL | global_asm!("", clobber_abi("C", FOO));
   |                                  ^^^ not a string literal

error: `clobber_abi` cannot be used with `global_asm!`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:129:19
   |
LL | global_asm!("{}", clobber_abi("C"), const FOO);
   |                   ^^^^^^^^^^^^^^^^

error: `clobber_abi` cannot be used with `global_asm!`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:131:28
   |
LL | global_asm!("", options(), clobber_abi("C"));
   |                            ^^^^^^^^^^^^^^^^

error: `clobber_abi` cannot be used with `global_asm!`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:133:30
   |
LL | global_asm!("{}", options(), clobber_abi("C"), const FOO);
   |                              ^^^^^^^^^^^^^^^^

error: `clobber_abi` cannot be used with `global_asm!`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:135:17
   |
LL | global_asm!("", clobber_abi("C"), clobber_abi("C"));
   |                 ^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^

error: duplicate argument named `a`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:137:35
   |
LL | global_asm!("{a}", a = const FOO, a = const BAR);
   |                    -------------  ^^^^^^^^^^^^^ duplicate argument
   |                    |
   |                    previously here

error: argument never used
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:137:35
   |
LL | global_asm!("{a}", a = const FOO, a = const BAR);
   |                                   ^^^^^^^^^^^^^ argument never used
   |
   = help: if this argument is intentionally unused, consider using it in an asm comment: `"/* {1} */"`

error: expected one of `#`, `clobber_abi`, `const`, `options`, or `sym`, found `""`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:140:28
   |
LL | global_asm!("", options(), "");
   |                            ^^ expected one of `#`, `clobber_abi`, `const`, `options`, or `sym`

error: expected one of `#`, `clobber_abi`, `const`, `options`, or `sym`, found `"{}"`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:142:30
   |
LL | global_asm!("{}", const FOO, "{}", const FOO);
   |                              ^^^^ expected one of `#`, `clobber_abi`, `const`, `options`, or `sym`

error: asm template must be a string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:144:13
   |
LL | global_asm!(format!("{{{}}}", 0), const FOO);
   |             ^^^^^^^^^^^^^^^^^^^^

error: asm template must be a string literal
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:146:20
   |
LL | global_asm!("{1}", format!("{{{}}}", 0), const FOO, const BAR);
   |                    ^^^^^^^^^^^^^^^^^^^^

error: the `in` operand cannot be used with `global_asm!`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:149:19
   |
LL | global_asm!("{}", in(reg));
   |                   ^^ the `in` operand is not meaningful for global-scoped inline assembly, remove it

error: the `out` operand cannot be used with `global_asm!`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:151:19
   |
LL | global_asm!("{}", out(reg));
   |                   ^^^ the `out` operand is not meaningful for global-scoped inline assembly, remove it

error: the `lateout` operand cannot be used with `global_asm!`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:153:19
   |
LL | global_asm!("{}", lateout(reg));
   |                   ^^^^^^^ the `lateout` operand is not meaningful for global-scoped inline assembly, remove it

error: the `inout` operand cannot be used with `global_asm!`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:155:19
   |
LL | global_asm!("{}", inout(reg));
   |                   ^^^^^ the `inout` operand is not meaningful for global-scoped inline assembly, remove it

error: the `inlateout` operand cannot be used with `global_asm!`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:157:19
   |
LL | global_asm!("{}", inlateout(reg));
   |                   ^^^^^^^^^ the `inlateout` operand is not meaningful for global-scoped inline assembly, remove it

error: the `label` operand cannot be used with `global_asm!`
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:159:19
   |
LL | global_asm!("{}", label(reg));
   |                   ^^^^^ the `label` operand is not meaningful for global-scoped inline assembly, remove it

error[E0435]: attempt to use a non-constant value in a constant
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:40:37
   |
LL |         asm!("{}", options(), const foo);
   |                                     ^^^ non-constant value
   |
help: consider using `const` instead of `let`
   |
LL -     let mut foo = 0;
LL +     const foo: /* Type */ = 0;
   |

error[E0435]: attempt to use a non-constant value in a constant
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:72:44
   |
LL |         asm!("{}", clobber_abi("C"), const foo);
   |                                            ^^^ non-constant value
   |
help: consider using `const` instead of `let`
   |
LL -     let mut foo = 0;
LL +     const foo: /* Type */ = 0;
   |

error[E0435]: attempt to use a non-constant value in a constant
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:75:55
   |
LL |         asm!("{}", options(), clobber_abi("C"), const foo);
   |                                                       ^^^ non-constant value
   |
help: consider using `const` instead of `let`
   |
LL -     let mut foo = 0;
LL +     const foo: /* Type */ = 0;
   |

error[E0435]: attempt to use a non-constant value in a constant
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:77:31
   |
LL |         asm!("{a}", a = const foo, a = const bar);
   |                               ^^^ non-constant value
   |
help: consider using `const` instead of `let`
   |
LL -     let mut foo = 0;
LL +     const foo: /* Type */ = 0;
   |

error[E0435]: attempt to use a non-constant value in a constant
##[error]  --> /checkout/tests/ui/asm/parse-error.rs:77:46
   |
LL |         asm!("{a}", a = const foo, a = const bar);
   |                                              ^^^ non-constant value
   |
help: consider using `const` instead of `let`
   |
LL -     let mut bar = 0;
LL +     const bar: /* Type */ = 0;
   |

error: aborting due to 70 previous errors

For more information about this error, try `rustc --explain E0435`.
```

</details>

--- a/compiler/rustc_parse/src/parser/diagnostics.rs
+++ b/compiler/rustc_parse/src/parser/diagnostics.rs
@@ -602,7 +602,9 @@ impl<'a> Parser<'a> {
         // Look for usages of '=>' where '>=' was probably intended
         if self.token == token::FatArrow
             && expected.iter().any(|tok| matches!(tok, TokenType::Operator | TokenType::Le))
-            && !expected.iter().any(|tok| matches!(tok, TokenType::FatArrow | TokenType::Comma))
+            && !expected
+                .iter()
+                .any(|tok| matches!(tok, TokenType::FatArrow | TokenType::CloseBrace))
         {
             err.span_suggestion(
                 self.token.span,

--- a/tests/ui/asm/parse-error.stderr
+++ b/tests/ui/asm/parse-error.stderr
@@ -57,6 +57,12 @@ error: expected one of `!`, `,`, `.`, `::`, `?`, `{`, or an operator, found `=>`
    |
 LL |         asm!("{}", in(reg) foo => bar);
    |                                ^^ expected one of 7 possible tokens
+   |
+help: you might have meant to write a "greater than or equal to" comparison
+   |
+LL -         asm!("{}", in(reg) foo => bar);
+LL +         asm!("{}", in(reg) foo >= bar);
+   |
 
 error: expected a path for argument to `sym`
   --> $DIR/parse-error.rs:32:24

--- a/tests/ui/parser/eq-gt-to-gt-eq.fixed
+++ b/tests/ui/parser/eq-gt-to-gt-eq.fixed
@@ -43,3 +43,9 @@ fn b() {
         _ => todo!(),
     }
 }
+
+fn closure() {
+    let a = 0;
+    let b = 1;
+    let _ = [a].iter().any(|x| x >= &b); //~ERROR
+}

--- a/tests/ui/parser/eq-gt-to-gt-eq.rs
+++ b/tests/ui/parser/eq-gt-to-gt-eq.rs
@@ -43,3 +43,9 @@ fn b() {
         _ => todo!(),
     }
 }
+
+fn closure() {
+    let a = 0;
+    let b = 1;
+    let _ = [a].iter().any(|x| x => &b); //~ERROR
+}

--- a/tests/ui/parser/eq-gt-to-gt-eq.stderr
+++ b/tests/ui/parser/eq-gt-to-gt-eq.stderr
@@ -109,5 +109,17 @@ LL -     match a => b {
 LL +     match a >= b {
    |
 
-error: aborting due to 7 previous errors
+error: expected one of `!`, `)`, `,`, `.`, `::`, `?`, `{`, or an operator, found `=>`
+  --> $DIR/eq-gt-to-gt-eq.rs:50:34
+   |
+LL |     let _ = [a].iter().any(|x| x => &b);
+   |                                  ^^ expected one of 8 possible tokens
+   |
+help: you might have meant to write a "greater than or equal to" comparison
+   |
+LL -     let _ = [a].iter().any(|x| x => &b);
+LL +     let _ = [a].iter().any(|x| x >= &b);
+   |
+
+error: aborting due to 8 previous errors
 

